src/pykeen/triples/triples_factory.py

```diff
@@ -730,7 +706,7 @@ def new_with_relations(
             f'keeping {len(relations)}/{self.num_relations} relations'
             f' and {mask.sum()}/{self.num_triples} triples in {self}',
         )
-        return self.with_other_triples(mapped_triples=self.mapped_triples[mask])
+        return self.clone_and_exchange_triples(mapped_triples=self.mapped_triples[mask])
```

Comment:
since almost all of the calls with this are using a mask, maybe you should even make a wrapper function that does this
```python
def with_mask(self, mask):
    return self.clone_and_exchange_triples(mapped_triples=self.mapped_triples[mask])
```

Reply:
Now it is only used twice: once with mask and once without :sweat_smile: